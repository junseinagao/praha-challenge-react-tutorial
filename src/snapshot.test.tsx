/// <reference types="vite/client"/>
import React from "react";
import { Meta, StoryFn } from "@storybook/react";
import { composeStories } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

type StoryFile = {
  default: Meta;
  [name: string]: StoryFn | Meta;
};

describe("Storybook Snapshots", async () => {
  const modulesByStoryName = await Promise.all(
    Object.entries(import.meta.glob<StoryFile>("./**/*.stories.tsx")).map(
      ([key, fn]) => {
        const chunks = key.split("/");
        const filename = chunks[chunks.length - 1];
        const storyName = filename.replace(/\.stories\..*/, "");

        return [storyName, fn()];
      }
    )
  );

  describe.each(modulesByStoryName.map(([name, module]) => ({ name, module })))(
    "$name",
    async ({ module }) => {
      test.each(
        Object.values(composeStories(await (module as Promise<StoryFile>))).map(
          (Story) => [Story.storyName!, Story]
        )
      )("%s", (_, Story) => {
        const { container } = render(<Story />);
        expect(container).toBeTruthy();
        expect(container).toMatchSnapshot();
      });
    }
  );
});
