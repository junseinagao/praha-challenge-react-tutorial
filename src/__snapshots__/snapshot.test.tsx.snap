// Vitest Snapshot v1

exports[`Storybook Snapshots > 'Board' > BoardWithAllTriangle 1`] = `
<div>
  <div
    class="status"
  >
    Winner: △
  </div>
  <div
    class="board-row"
  >
    <button
      class="square"
    >
      △
    </button>
    <button
      class="square"
    >
      △
    </button>
    <button
      class="square"
    >
      △
    </button>
  </div>
  <div
    class="board-row"
  >
    <button
      class="square"
    >
      △
    </button>
    <button
      class="square"
    >
      △
    </button>
    <button
      class="square"
    >
      △
    </button>
  </div>
  <div
    class="board-row"
  >
    <button
      class="square"
    >
      △
    </button>
    <button
      class="square"
    >
      △
    </button>
    <button
      class="square"
    >
      △
    </button>
  </div>
</div>
`;

exports[`Storybook Snapshots > 'Game' > GameWithNoProps 1`] = `
<div>
  <div
    class="game"
  >
    <div
      class="game-board"
    >
      <div
        class="status"
      >
        次のプレイヤー: X
      </div>
      <div
        class="board-row"
      >
        <button
          class="square"
        />
        <button
          class="square"
        />
        <button
          class="square"
        />
      </div>
      <div
        class="board-row"
      >
        <button
          class="square"
        />
        <button
          class="square"
        />
        <button
          class="square"
        />
      </div>
      <div
        class="board-row"
      >
        <button
          class="square"
        />
        <button
          class="square"
        />
        <button
          class="square"
        />
      </div>
    </div>
    <div
      class="game-info"
    >
      <ol>
        <li>
          <button>
            Go to game start
          </button>
        </li>
      </ol>
    </div>
  </div>
</div>
`;

exports[`Storybook Snapshots > 'Square' > SquareWithO 1`] = `
<div>
  <button
    class="square"
  >
    O
  </button>
</div>
`;

exports[`Storybook Snapshots > 'Square' > SquareWithTriangle 1`] = `
<div>
  <button
    class="square"
  >
    △
  </button>
</div>
`;

exports[`Storybook Snapshots > 'Square' > SquareWithX 1`] = `
<div>
  <button
    class="square"
  >
    X
  </button>
</div>
`;
